jfix-stdlib-concurrency/src/main/java/ru/fix/stdlib/concurrency/threads/ReschedulableScheduler.java

```diff
@@ -45,41 +51,64 @@ private void detachIndicators() {
         profiler.detachIndicator(scheduledTasksIndicatorName);
     }
 
+    /**
+     * Same as {@link #schedule(DynamicProperty, DynamicProperty, Runnable)}
+     * but doesn't throw exception if {@link ReschedulableScheduler#shutdown()} was executed
+     *
+     * @return result task from executionService or null if {@link ReschedulableScheduler#shutdown()} was executed
+     */
+    public ScheduledFuture<?> scheduleIfNotShutdown(DynamicProperty<Schedule> scheduleSupplier,
+                                                    DynamicProperty<Long> startDelay,
+                                                    Runnable task) {
+        if (!shutdownReadLock.tryLock() || isShutdown) {
+            return null;
+        }
```

Comment:
Error prone code.  
```
suppose
shutdownReadLock.tryLock() returns true
and isShutdown is false

        if (!shutdownReadLock.tryLock() || isShutdown) {
            return null;
        }

we will take lock and not release it
```
I see that shutdownReadLock and isShutdown should be consistent between each other. But this thing is hard to understand / complex to remember and creates unnecessary coupling.
Lets split responsibility: locks - responsible for access, isShutdown - responsible for state. 
Always release locks in final block regardless of other code behavior, expected or not. 
```
lock  = shutdownReadLock.lock()
try  {  ... isShutdown... } finally {lock.unlock() }

```